servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -1141,6 +1148,40 @@ private void propagatePersisted(Inode inode, boolean replayed)
   }
 
   /**
+   * Create the permissionStatus for file or directory.
+   * @param remote true if the request is for creating permission from client side, the
+   * username binding into inode will be gotten from {@code AuthorizedClientUser.get().getName()}.
+   * If the remote is false, the username binding into inode will be gotten from {@link LoginUser}
+   * @return permissionStatus
+   * @throws IOException
+   */
+  private PermissionStatus createPermissionStatus(boolean remote) {
```

Comment:
Do you think this method belongs to this file? It seems quite generic and it doesn't depend on any state of `FileSystemMaster`.